autoload/neomake/makers/ft/php.vim

```diff
@@ -42,8 +42,19 @@ function! neomake#makers#ft#php#phpmd() abort
 endfunction
 
 function! neomake#makers#ft#php#phpstan() abort
-    return {
-        \ 'args': ['analyse', '--errorFormat', 'raw'],
+    " PHPStan normally considers 0 to be the default level, so that is used here as the default:
+    let maker = {
+        \ 'args': ['analyse', '--errorFormat', 'raw', '--level', get(g:, 'neomake_phpstan_level', 0)],
         \ 'errorformat': '%E%f:%l:%m',
         \ }
+    " Check for the existence of a default PHPStan project configuration file.
+    " Technically PHPStan does not have a concept of a default filename for a
+    " project configuration file, but phpstan.neon is the filename shown in the
+    " example in the PHPStan documentation, so this is the default name expected
+    " by Neomake.
+    let phpStanConfigFilePath = neomake#utils#FindGlobFile('phpstan.neon')
+    if !empty(phpStanConfigFilePath)
+        call extend(maker.args, ['-c', phpStanConfigFilePath])
+    endif
+    return maker
```

Comment:
I'm not sure if the config file has a level option or not, but it is definitely worth finding out. Even if it did, I'm not sure which one would have higher precedence. I will try to find the answer to that question. For now I just know that the --level option becomes required when you do specify a config. I'm not opposed to just letting people specify it in the config instead if that works. In that case it would make sense to pass a value of 0 when it becomes a required flag.